docs/rfcs/GPUXDialect.md

```diff
@@ -0,0 +1,233 @@
+# RFC: GPUX Dialect (Extension of upstream GPU Dialect)
+
+Core MLIR Team
+
+## Summary
+
+We propose the GPUX dialect as an extension of the upstream GPU dialect. The GPUX dialect allows us to expose stream creation/destruction ops.These ops are required by the underlying runtimes (Level zero & Sycl) for explicit stream/context/device creation.
+The GPUX dialect will also have some of the upstream GPU dialect ops extended with an added argument for stream in those ops.
+
+## Motivation
+
+Upstream MLIR has a Dialect called the [GPU Dialect](https://mlir.llvm.org/docs/Dialects/GPU/) which provides middle-level abstractions for launching GPU kernels following a programming model similar to that of CUDA or OpenCL. The upstream dialect exposes all the operation required to launch the kernel on a GPU device. But, the drawback with the upstream dialect is it does not expose Stream as an operation in the dialect. For example, if the user wants to launch the kernel on a particular stream(provided by the user), there is no way to do that today. For the upstream dialect, stream is created internally while lowering to LLVM runtime calls and it refers specifically to the CUDA stream. Therefore, the need to create a GPUX Dialect arises. The GPUX dialect is an extension of the upstream GPU dialect with following ops - CreateStreamOp, DestroyStreamOp, LaunchFuncOp, AllocOp, DeallocOp & WaitOp . The CreateStreamOp op allows for explicit Stream creation. The stream is a temporary stream and is internally creating a SYCL/L0 queue, with default context and device.
```

Comment:
Although you have already mentioned new ops in the summary above, pointing out which ops are new and which ops are an extension of an existing GPU dialect op would help readers get a better picture.